test/unit/puppetclass_test.rb

```diff
@@ -246,4 +246,32 @@ def check_object_config_group(obj)
       end
     end
   end
+
+  context "going down the rabbit hole" do
+    test "three levels of nested attributes still validate nested nested nested objects" do
+      klass = FactoryGirl.create(:puppetclass)
+      hostgroup = FactoryGirl.create(:hostgroup)
+      lk = FactoryGirl.create(:variable_lookup_key, puppetclass_id: klass.id)
+      attributes = {"hostgroup_ids"=>[hostgroup.id],
+        "lookup_keys_attributes"=>
+        {"0"=>
+          { "_destroy"=>"false",
+            "key"=>"hahs",
+            "description"=>"",
+            "key_type"=>"hash",
+            "default_value"=>"{\"foo\" => \"bar\"}",
```

Comment:
Is this an invalid default value? Looks like a valid hash. I'm asking because I think you're trying to reproduce what you describe on http://projects.theforeman.org/issues/12805